### PR TITLE
test/recipes: split 81_test_cmp_cli.t, add test using -engine loader_attic

### PR DESCRIPTION
**Pull request by DDvO**
*  Add a new test case to 81_test_cmp_cli.t, testing engine support using `-engine loader_attic`.

* Move the HTTP-based CMP tests to 80_test_cmp_http.t, to start a little earlier.
This should decrease total test run time due to better parallelization.

* `openssl.pod`: Add documentation for using the loader_attic engine

* `apps/cmp.c`: Correct `-keyform` option range w.r.t engine

This PR also includes a tweak to `run_tests.pl`: Improve diagnostics on the use of `HARNESS_JOBS`.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
Could you please rebase to fix the conflicts?

**Comment by DDvO** [comment]
Nice that you are handling this PR now.

> Could you please rebase to fix the conflicts?

Done. Interestingly, `git rebase` was able to handle them automatically, 
likely because the "conflicts" that had been reported above were just due to 
changes to test data files that ware just moved in this PR without changing them.

File: doc/man1/openssl.pod [hunk 1]
```diff
@@ -615,6 +615,11 @@ L<config(5)/Engine Configuration>.
 The engine will be used for key ids specified with B<-key> and similar
 options when an option like B<-keyform engine> is given.
 
+A special case is the C<loader_attic> engine, which supports loading keys,
+parameters, certificates, and CRLs from files.
+When this engine is used, files with such credentials are read via this engine.
+Using the C<file:> schema is optional; a plain file (path) name will do.
```
Comment (t8m) [inline doc/man1/openssl.pod:621]: This engine is IMO useful just for internal openssl testing purposes. So perhaps add some note about that?

Comment (DDvO) [reply doc/man1/openssl.pod:621]: Good point; done.

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Merged - thanks @t8m

**Pull request closed without merging**